### PR TITLE
Add Crin PCB to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding Crin PCB to QMK

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/crin/halconf.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *    `qmk chibios-confmigrate -i keyboards/cannonkeys/onyx/halconf.h -r platforms/chibios/common/configs/halconf.h`
+ */
+
+#pragma once
+
+#define HAL_USE_I2C TRUE
+
+#define HAL_USE_PWM TRUE
+
+#define HAL_USE_SPI TRUE
```
Comment (zvecr) [inline keyboards/cannonkeys/crin/halconf.h:28]:
```suggestion
#define HAL_USE_PWM TRUE
```

File: keyboards/cannonkeys/crin/mcuconf.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#include_next <mcuconf.h>
+
+#undef STM32_I2C_USE_I2C1
+#define STM32_I2C_USE_I2C1 TRUE
+
+#undef STM32_PWM_USE_TIM3
+#define STM32_PWM_USE_TIM3 TRUE
+
+#undef STM32_SPI_USE_SPI2
+#define STM32_SPI_USE_SPI2 TRUE
```
Comment (zvecr) [inline keyboards/cannonkeys/crin/mcuconf.h:33]:
```suggestion
#undef STM32_PWM_USE_TIM3
#define STM32_PWM_USE_TIM3 TRUE
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by awkannan** [comment]
@zvecr Thank you for the catch!

**Comment by awkannan** [comment]
@drashna This one is a bit weird, as zvecr's suggested changes did not make it in before the prior PR was merged.
I think we should still merge these.

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**